build: Delete unused files

selliott512 opened the pull request:
Delete files that have no impact on the three IWADs built by searching
for references to them. Deleting these files has no impact on the
three IWADs built so long as "VERSION" in "Makefile" is changed to a
static string.

----

I thought I'd take a shot at deleting old unused files. If some of them are
around for historical or documentation reasons I can add them back.

Incidentally, I was inspired to do this by the spaces in the "patches/agm_logo"
files. Being an old fashioned *nix type I was curious if such a file was really
needed.

I verified that the three IWADs were not impacted by these changes by first
changing the version that appears on the title screen to a static string:
``` diff
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
-VERSION=$(shell git describe --abbrev=8 --dirty 2>/dev/null || echo v0.12.1)
+VERSION=ver-stub
 WADS=wads
 ASCIIDOC=asciidoc
 ADOCOPTS=--backend=html5 --conf-file=.adoc-layout.conf
```

And then doing a clean build, with and without this change, followed by
checking the hashes:
``` shell
git clean -fxd && make && sha1sum ~/git/freedoom/wads/*.wad
```

fragglet commented on patches/mc9.png:
We have mc1-20, this seems like an oversight. Add to textures config?

Talon1024 replied:
There is no MC9 patch in plutonia.wad

fragglet reviewed (changes requested):
I'm fine with most of these. A few exceptions though

selliott512 commented:
I've updated this to latest master and verified that it still no longer impacts the IWADS produced (by the method I mentioned in a comment).

selliott512 commented:
@fragglet Is there anything else I can do for this? I was recently bothered by the files with spaces in their paths, which is why I thought of this again.

As to `patches/aqpipe06.png` here's what the surrounding images look like:

Three images before patches/aqpipe06.png (03 to 05):
![aqpipe03](https://github.com/freedoom/freedoom/assets/2380178/6419a019-310c-4a0c-90ad-2907628b245a) ![aqpipe04](https://github.com/freedoom/freedoom/assets/2380178/b8f694ea-56d0-4bf4-bc43-aa86c68c46db) ![aqpipe05](https://github.com/freedoom/freedoom/assets/2380178/7dcbeffc-a466-4edf-8b79-015988c23528)

patches/aqpipe06.png:
![aqpipe06](https://github.com/freedoom/freedoom/assets/2380178/33f29a77-d794-4c23-8d07-77bd0d01ce3c)

Three images after patches/aqpipe06.png (07 to 09):
![aqpipe07](https://github.com/freedoom/freedoom/assets/2380178/a4a18d48-c85d-4fe3-b23e-119d706dd5c6)
![aqpipe08](https://github.com/freedoom/freedoom/assets/2380178/45372237-aa18-4a29-a1de-f26b1c36633a)
![aqpipe09](https://github.com/freedoom/freedoom/assets/2380178/aec8bba6-7fd3-4c22-bc4b-48d58c2f7034)

It's a transition from vertical pipes to horizontal textures. Is `patches/aqpipe07.png` intended to be used instead? I'm happy to add it to `lumps/textures/textures.cfg` instead of deleting it, if you want..

mc776 commented:
I think AQPIPE06, AQPIPE07 and BROWN4 should be kept, as per #1061.

selliott512 commented:
> I think AQPIPE06, AQPIPE07 and BROWN4 should be kept, as per #1061.

Good catch. This would've caused a build failure once your PR to add those sprites had been merged, so it's nice to catch it before then.

Occasionally I build the WADs with and without these deletes to make sure the WADs are not impacted. I'll do that again, maybe this weekend.

Since this PR will be four years old this year I don't expect it to be merged any time soon. Then again, things have been pretty active recently, so who knows?

selliott512 commented:
I have ...

- Updated to latest master.
- Confirmed that the deletion of the three patches that mc776 mentioned results in an error (because they have since been added to `lumps/textures/textures.cfg`).
- Restored the three patches (they are no longer deleted by this MR).
- Verified that this MR does not impact that WADs produced as per my first comment.

I think I've addressed all per file controversies other than possibly `patches/mc9.png`.

mc776 commented on sounds/dspedeth.wav:
Apparently because of this typo the matribite/summoner never had any death sound at all, despite this being right here, between 2007 and 2021 :| 

Honestly this sounds better than the current dspedth - it just needs an explosion, and shortening so it isn't cut off when the actor disappears (which is a problem with the current one as well).

Maybe let's delete this now, but I'll keep a copy and see if we can mix something new.

mc776 replied:
Author is undead003

mc776 replied:
posted to #1102

mc776 commented:
> I think I've addressed all per file controversies other than possibly `patches/mc9.png`
For ease of reference, this is mc9:
![mc9](https://github.com/freedoom/freedoom/assets/24984517/0d7d8d07-3c86-4555-99c8-e3ad6fb2a841)

It's a *very* primitive lava texture with a big lump that repeats over and over again. I literally cannot imagine any use for it.

I think this PR is ready to merge.

selliott512 commented:
Is there any chance of this ever being approved? I would like to at least get rid of the files that have spaces in their names, but I could just close this instead of having it sit as an open PR indefinitely.
